src/components/common/FileUploadPdf.vue

```diff
@@ -0,0 +1,297 @@
+<template>
+  <v-file-input
+    id="file-upload-pdf"
+    dense
+    filled
+    show-size
+    color="primary"
+    hide-details="auto"
+    accept=".pdf"
+    label="Select a file to upload"
+    :value="file"
+    :key="count"
+    :error-messages="errorMessages"
+    @change="onChange($event)"
+  />
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { AxiosResponse } from 'axios'
+import { Component, Emit, Prop } from 'vue-property-decorator'
+import { GlobalWorkerOptions, getDocument } from 'pdfjs-dist/build/pdf'
+import { PageSizes, PAGE_SIZE_DICT } from '@/enums'
+import { PdfInfoIF, PresignedUrlIF } from '@/interfaces'
+
+// set web worker
+GlobalWorkerOptions.workerSrc = require('pdfjs-dist/build/pdf.worker.entry')
+
+@Component({})
+export default class FileUploadPdf extends Vue {
+  @Prop({ default: null }) readonly file!: File
+  @Prop({ default: null }) readonly fileKey!: string
+  @Prop({ default: true }) readonly isRequired!: boolean
+  @Prop({ default: 0 }) readonly maxSize!: number // in MB
+  @Prop({ default: null }) readonly pageSize!: PageSizes
+  @Prop({ required: true }) readonly userId!: string
+
+  // Network service functions
+  @Prop({ required: true })
+  readonly getPresignedUrl!: (fileName: string) => Promise<PresignedUrlIF>
+  @Prop({ required: true })
+  readonly uploadToUrl!: (url: string, file: File, key: string, userId: string) => Promise<AxiosResponse>
```

Comment:
I'm passing these in for when this component is shared -- each parent can provide these based on their local services, so THIS component doesn't need to know about the network.